File: backend/Services/Parsers/DungleonParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class DungleonParser : BasicScoreResultParser
    {
        private ILogger<DungleonParser> _logger;

        public DungleonParser(ILogger<DungleonParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Dungleon";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{Url} #\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => null;
        public override string Url => "https://www.dungleon.com";
        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace(Url, GameName).Trim();
        }
    }
}

File: lambdas/src/parsers/dungleon-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';

export class DungleonParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Dungleon';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://www.dungleon.com';
  
  // Matches "https://www.dungleon.com #850 X/6" (share format) or "Dungleon #850 X/6" (cleaned format)
  // C# uses: `{Url} #\d+ (?<score>[\d|X])/\d` - only matches URL format
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `(?:${this.url}|${this.gameName}) #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result.replace(this.url, this.gameName).trim();
  }
}
